### PR TITLE
[DOP-7289] Implement KafkaProtocol

### PR DESCRIPTION
**Pull request by dmitry-pedchenko**
## Change Summary

Implement KafkaProtocol and tests for it

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [x] Documentation reflects the changes where applicable
* [x] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/onetl/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: onetl/connection/db_connection/kafka/connection.py [hunk 1]
```diff
@@ -68,21 +72,16 @@ class Kafka(DBConnection):
         An attribute that contains a class that generates a Kafka connection configuration.
         It depends on the type of connection to Kafka.
 
-    .. warning::
-
-        When creating a connector, when specifying `user` parameter, either `password` or `keytab` can be specified. Or
-        these parameters for anonymous connection are not specified at all.
+    protocol : IKafkaProtocol, default: ``PlaintextProtocol``
+        Class containing connection parameters. If the protocol parameter is not specified, then the parameter will be
+        passed `PLAINTEXT`, otherwise the `SASL_PLAINTEXT` parameter will be passed to the `kafka.security.protocol`
```
Comment (dolfinus) [inline onetl/connection/db_connection/kafka/connection.py:77]: Замени одинарный backtick на двойные

File: onetl/connection/db_connection/kafka/connection.py [hunk 2]
```diff
@@ -121,6 +120,8 @@ class Kafka(DBConnection):
     ReadOptions = KafkaReadOptions
     WriteOptions = KafkaWriteOptions
     Dialect = KafkaDialect
+    PlaintextProtocol = KafkaPlaintextProtocol
+    protocol: IKafkaProtocol = PlaintextProtocol()
```
Comment (dolfinus) [inline onetl/connection/db_connection/kafka/connection.py:124]: Перенеси после `auth`

File: onetl/connection/db_connection/kafka/ikafka_protocol.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from onetl.connection import Kafka
+
+
+class IKafkaProtocol(ABC):
```
Comment (dolfinus) [inline onetl/connection/db_connection/kafka/ikafka_protocol.py:23]: Все же предлагаю использовать одинаковый нейминг в репозитории, BaseKafkaProtocol.

Comment (dolfinus) [reply onetl/connection/db_connection/kafka/ikafka_protocol.py]: И с IKafkaAuth тоже самое

File: onetl/connection/db_connection/kafka/connection.py (hunk 2 above)
Comment (dolfinus) [reply onetl/connection/db_connection/kafka/connection.py]: Я про поле `auth`. `PlaintextProtocol` пусть будет рядом с остальными классами

**Review by dolfinus — APPROVED** [review]

**Review by maxim-lixakov — APPROVED** [review]

**Pull request merged**